sql/core.go

```diff
@@ -73,6 +73,13 @@ type FunctionExpression interface {
 	FunctionName() string
 }
 
+// WindowExpression is an Expression that represents a window aggregation
+type WindowExpression interface {
```

Comment:
Doesn't seem like either of these new methods really needs a public interface for the time being, maybe hold off on this for now?

Reply:
Ok, I'll need to figure something else out for the helper functions to be common. Maybe type switch verify that the window field is valid.